### PR TITLE
added methods to endpoint, added model info to endpoint, added card v…

### PR DESCRIPTION
**Pull request by mackmc22**
I can't remember what we were doing with the id getting passed in, assuming that would get passed in as the username? Also having a difficult time figuring out how I get the card that is dealt to the appropriate hand. Not sure I entirely understand the connection with the model (or I probably forgot). This seems beyond the functionality of the librarian endpoints that I'm referencing, but I could be wrong. Either way, need some help on this.

### CODE REVIEW

File: blackjack/api/views.py [hunk 1]
```diff
@@ -20,8 +20,17 @@ def __init__(self):
...
+    player_hand = data['player_hand']
+    dealer_hand = data['dealer-hand']
 
+    # how do I get this into the hand that called the function?
+    card_dealt = Game.deck_cards.pop()
 
-@api_view(["GET"])
-def deal(request, id):
-    return JsonResponse(data={'card': 2}, status=status.HTTP_200_OK)
+    GameState.objects.create(username=username, deck=deck, player_hand=player_hand, dealer_hand=dealer_hand)
```
Comment (de1ux) [inline blackjack/api/views.py]: you just dealt a card (`card_dealt`). what does that mean for the contents of the `deck` variable, and the `player_hand` variable?

File: blackjack/api/views.py [hunk 2]
```diff
@@ -20,8 +20,17 @@ def __init__(self):
         random.shuffle(self.deck_cards)
 
 
+@api_view(["GET", "POST", "PUT"])
+def deal(request, id):
+    data = json.loads(request.body)
```
Comment (de1ux) [inline blackjack/api/views.py]: good intuition (we will want to read some data from the `request`, eventually)

for now, ignore the `request` object and manually assign values to these variables from the Game instance

File: blackjack/api/views.py [hunk 3]
```diff
@@ -20,8 +20,17 @@ def __init__(self):
...
+    username = data['username']
+    deck = data['deck']
+    player_hand = data['player_hand']
+    dealer_hand = data['dealer-hand']
 
+    # how do I get this into the hand that called the function?
+    card_dealt = Game.deck_cards.pop()
```
Comment (de1ux) [inline blackjack/api/views.py]: dont forget to create an instance of the `Game` class! don't want to use the class directly

Comment (mackmc22) [reply blackjack/api/views.py]: I remember that now, I think I fixed it

**Pull request merged**